### PR TITLE
Cloud endpoints docs v1

### PR DESCRIPTION
**Pull request by ForgoneReality**

### CODE REVIEW

File: docs/network-edge/cloud-endpoints.mdx [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+- Use the [Kubernetes Ingress Controller](/k8s/)
+
+When you create a Cloud Endpoint, you must specify:
+
+- The complete URL for the cloud endpoint, including the scheme, hostname, and port.
+  The port may be inferred if omitted
+- The associated traffic policy
+- The bindings [LINK?] associated with the cloud endpoint
+
+Cloud endpoints are commonly used route traffic to agent endpoints (endpoints initiated by the ngrok agent) via the
```
Comment (DevMandy) [inline docs/network-edge/cloud-endpoints.mdx:40]: Cloud endpoints are commonly used to route traffic to agent endpoints

File: docs/network-edge/cloud-endpoints.mdx [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+custom-response, or forward via the forward action.
+
+For examples of cloud endpoint usage, check out the following guides:
+
+- [Forwarding Traffic to and Load Balancing Internal Endpoints with Cloud Endpoints](/guides/other-guides/forwarding-and-load-balancing-with-cloud-endpoints/)
+- [Path-based Routing and Policy Decentralization with Cloud Endpoints](/guides/other-guides/path-based-routing-and-policy-decentralization-with-cloud-endpoints/)
+
+## Cloud Endpoints vs Edges
+
+Cloud endpoints are the new edges. Cloud endpoints offer all the functionality of edges and simplifies the module
```
Comment (DevMandy) [inline docs/network-edge/cloud-endpoints.mdx:52]: Cloud endpoints offer all the functionality of edges and simplify the module

File: docs/guides/other-guides/path-based-routing-and-policy-decentralization-with-cloud-endpoints.mdx [hunk 3]
```diff
@@ -0,0 +1,260 @@
...
+The binding is set to public so anyone can access our cloud endpoint.
+
+```bash
+ngrok api endpoints create \
+    --api-key YOUR_NGROK_API_KEY \
+    --bindings public \
+    --description "sample description" \
+    --metadata sample-metadata \
+    --url https://${NGROK_SUBDOMAIN}.ngrok.app
+    --traffic-policy '{"on_http_request":[{"expressions":["req.url.path.startsWith('/api/')"],"actions":[{"type":"forward-internal","config":{"url":"https://{YOUR_API_DOMAIN}.internal"}}]},{"actions":[{"type":"forward-internal","config":{"url":"https://{YOUR_HOMEPAGE_DOMAIN}.internal"}}]}]}'
```
Comment (euank) [inline docs/guides/other-guides/path-based-routing-and-policy-decentralization-with-cloud-endpoints.mdx:237]:
```suggestion
    --traffic-policy "$(cat traffic-policy.yml)"
```

Or:

```suggestion
--traffic-policy "$(cat <<EOF
on_http_request:
# etc etc
EOF
)"
```

We really need to make `api` take a `--traffic-policy-file` flag too.

**Review by euank — COMMENTED** [review]
Thank you for writing all this, overall looks like very helpful content to have!

I have a lot of small wording nits and tweaks for the load-balcing one, and a few for the path-based-routing one.
Since there's a lot of snippets that are shared by both, I didn't comment on both, I figure they'll both get updates similarly

File: docs/guides/other-guides/forwarding-and-load-balancing-with-cloud-endpoints.mdx [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+
+Let's start by creating a internal endpoint, initiated from the agent.
+The endpoints will be in HTTP for this guide, but you can also use TCP or TLS.
+
+```bash
+ngrok http 80 \
+    --url https://${YOUR_DOMAIN_NAME}.internal
+```
+
+- Replace or set `YOUR_DOMAIN_NAME` as the domain you'd like to use for this guide. You can set it to anything you want, so long as it isn't taken already!
```
[the remark this answers is not in the document]
Comment (ForgoneReality) [reply docs/guides/other-guides/forwarding-and-load-balancing-with-cloud-endpoints.mdx:57]: > This implies internal domains are globally unique, but they are'nt.

was not aware, thank you!!

**Review by jmhertlein — APPROVED** [review]

**Pull request merged**